forum/users/views.py

```diff
@@ -1,3 +1,52 @@
-from django.shortcuts import render
+import logging
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from .serializers import CustomUserSerializer
+from rest_framework.permissions import AllowAny
 
-# Create your views here.
+logger = logging.getLogger('users')
+
+class RegisterUserView(APIView):
+    """
+    View for user registration.
+    """
+    permission_classes = [AllowAny]
+
+    def post(self, request):
+
+        # Pass the incoming data to the serializer
+        logger.info("User registration request received.")
+        serializer = CustomUserSerializer(data=request.data)
+
+        # Validate the data
+        if serializer.is_valid():
+            try:
+                user = serializer.save()
+
+                # Return a response with limited user data
+                logger.info(
+                        f"User successfully registered. User ID: {user.user_id}, Email: {user.email}")
+                
+                response_data = {
+                    "user_id": user.user_id,
+                    "email": user.email,
+                    "first_name": user.first_name,
+                    "last_name": user.last_name,
+                    "role": user.get_role_display(),
+                    "created_at": user.created_at,
+                }
+                return Response(response_data, status=status.HTTP_201_CREATED)
+            
+            except Exception as e:
+                logger.error(f"Error during user registration: {e}", exc_info=True)
+                return Response(
```

Comment:
You might want to consider propagating general error further as it might be useful to also let django handle it: 
```python
try:
    ...
except APIException as e:
    logger.error(f"API error during signup: {e}")
    raise
except Exception as e:
    logger.critical(f"Unexpected error during signup: {e}", exc_info=True)
    raise
```
Unless you are also handling some specific error, info about which might be informative to user:
```python
try:
    user = serializer.save()
except IntegrityError as e:
    logger.warning(f"Signup failed due to integrity error: {e}")
    return Response(
        {"detail": "A user with this email already exists."},
        status=status.HTTP_400_BAD_REQUEST,
    )
```

Reply:
adjusted logger and added dynamic __name__